testing/trino-product-tests/src/main/java/io/trino/tests/product/deltalake/TestDeltaLakeAlluxioCaching.java

```diff
@@ -0,0 +1,94 @@
+import static org.testng.Assert.assertEquals;
+
+public class TestDeltaLakeAlluxioCaching
+        extends ProductTest
+{
+    @Test(groups = {DELTA_LAKE_ALLUXIO_CACHING, PROFILE_SPECIFIC_TESTS})
+    public void testReadFromCache()
+    {
+        testReadFromTable("table1");
+        testReadFromTable("table2");
+    }
+
+    private void testReadFromTable(String tableNameSuffix)
+    {
+        String cachedTableName = "delta.default.test_cache_read" + tableNameSuffix;
+        String nonCachedTableName = "delta_non_cached.default.test_cache_read" + tableNameSuffix;
+
+        int rowCount = createTestTable(cachedTableName);
+
+        CacheStats beforeCacheStats = getCacheStats("delta");
+
+        long tableSize = (Long) onTrino().executeQuery("SELECT SUM(size) as size FROM (SELECT \"$path\", \"$file_size\" AS size FROM " + nonCachedTableName + " GROUP BY 1, 2)").getOnlyValue();
+
+        assertThat(onTrino().executeQuery("SELECT * FROM " + cachedTableName))
+                .hasRowsCount(rowCount);
+
+        assertEventually(
+                new Duration(20, SECONDS),
+                () -> {
+                    // first query via caching catalog should fetch external data
+                    CacheStats afterQueryCacheStats = getCacheStats("delta");
+                    assertGreaterThanOrEqual(afterQueryCacheStats.cacheSpaceUsed(), beforeCacheStats.cacheSpaceUsed() + tableSize);
+                    assertGreaterThan(afterQueryCacheStats.externalReads(), beforeCacheStats.externalReads());
+                    assertGreaterThanOrEqual(afterQueryCacheStats.cacheReads(), beforeCacheStats.cacheReads());
+                });
+
+        assertEventually(
+                new Duration(10, SECONDS),
+                () -> {
+                    CacheStats beforeQueryCacheStats = getCacheStats("delta");
+
+                    assertThat(onTrino().executeQuery("SELECT * FROM " + cachedTableName))
+                            .hasRowsCount(rowCount);
+
+                    // query via caching catalog should read exclusively from cache
+                    CacheStats afterQueryCacheStats = getCacheStats("delta");
+                    assertGreaterThan(afterQueryCacheStats.cacheReads(), beforeQueryCacheStats.cacheReads());
+                    assertEquals(afterQueryCacheStats.externalReads(), beforeQueryCacheStats.externalReads());
+                    assertEquals(afterQueryCacheStats.cacheSpaceUsed(), beforeQueryCacheStats.cacheSpaceUsed());
+                });
+
+        onTrino().executeQuery("DROP TABLE " + nonCachedTableName);
+    }
+
+    /**
+     * Creates a table which should contain two ~2 MB parquet files
+     */
+    private int createTestTable(String tableName)
+    {
+        int limit = 150_000;
+        onTrino().executeQuery("DROP TABLE IF EXISTS " + tableName);
+        onTrino().executeQuery("CREATE TABLE " + tableName + " AS SELECT * FROM tpch.sf1.customer LIMIT " + limit);
```

Comment:
Instead of using LIMIT, you can use catalog session property `target_max_file_size` to cap max file size